internal/Get-Connection.ps1

```diff
@@ -0,0 +1,115 @@
+﻿Function Get-Connection
```

Comment:
In-Test-sqlconnection I would bundle all there is to know about settings and requirements for a connection. So, client side info can be OS, PSVersion, elevated rights, smo installed,.... and then server side, and indeed domain and network items if they are of concern. I would include Kerberos info too.

I don't understand what you mean by the second remark. You prefer separate internal Test- functions or one containing both ping and CIM?